pkg/cosign/verify_test.go

```diff
@@ -1228,6 +1228,37 @@ func TestValidateAndUnpackCertWithIdentities(t *testing.T) {
 		}
 	}
 }
+
+func TestValidateAndUnpackCertWithIntermediatesSuccess(t *testing.T) {
+	subject := "email@email"
+	oidcIssuer := "https://accounts.google.com"
+
+	rootCert, rootKey, _ := test.GenerateRootCa()
+	subCert, subKey, _ := test.GenerateSubordinateCa(rootCert, rootKey)
+	leafCert, _, _ := test.GenerateLeafCert(subject, oidcIssuer, subCert, subKey)
+
+	rootPool := x509.NewCertPool()
+	rootPool.AddCert(rootCert)
+	subPool := x509.NewCertPool()
+	rootPool.AddCert(subCert)
```

Comment:
```suggestion
```

Reply:
Sorry one little thing, this should be removed, since you're testing that a chain can be built with the intermediate in the subPool to the root in the rootPool